src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -490,26 +501,16 @@ void RayletBasedActorScheduler::HandleWorkerLeaseReply(
       // The runtime environment has failed by an unrecoverable error.
       // We cannot create this actor anymore.
       if (reply.runtime_env_setup_failed()) {
-        // Right now, the way to report error message back to actor is not that great.
-        // This message is used to notify users the cause of actor death (with ERROR
-        // severity).
-        // TODO(sang): It is a temporary solution. Improve the error reporting here.
-        RAY_LOG(ERROR)
-            << "Cannot create an actor "
-            << actor->GetCreationTaskSpecification().FunctionDescriptor()->CallString()
-            << " of an id " << actor->GetActorID()
-            << " because the runtime environment setup failed on a node of address "
-            << node->node_manager_address() << ".";
-        RAY_LOG(INFO) << "Actor failed to be scheduled on a node " << node_id;
-        schedule_failure_handler_(std::move(actor), /*destroy_actor*/ true);
+        HandleRuntimeEnvSetupFailure(actor, node_id);
```

Comment:
Can we just not write a method for this? I think it is easier to read that it is in here (we don't need to follow the function def and this method is not resuable)

Reply:
I abstract this function because this part of code is duplicate in `GcsBasedActorScheduler` and `RayletBasedActorScheduler`